localMatchApp/sources/Login.cpp

```diff
@@ -0,0 +1,82 @@
+#include "Login.h"
+#include <iostream>
+
+using namespace std;
+
+Login::Login ( ) {
+  // ToDo: Connect to database.
+  // This console out is created for debug process
+  std::cout<<"object created"<<std::endl;
+}
+
+Login::~Login ( ) {
+  // ToDo: Disconnect to database.
+  // This console out is created for debug process
+  std::cout<<"object deleted"<<std::endl;
+}
+
+Login::ConnectionMessage Login::signUp ( const string& i_user, const string& i_password,
+                                         const string& i_repeatedPassword ) {
+  ConnectionMessage signUpConnectionState=ConnectionMessage::goodEntry;
+  // ToDo: Send and recive information to database.
+  // This string is just created for debug process
+  string userInDatabase = "Lol2018";
+
+  if ( !hasCorrectFormat ( i_password ) || !hasCorrectFormat ( i_repeatedPassword ) ) {
+    signUpConnectionState= ConnectionMessage::wrongPassword;
+  }
+
+  if ( i_password != i_repeatedPassword && (hasCorrectFormat ( i_password ) && hasCorrectFormat ( i_repeatedPassword ))) {
+    signUpConnectionState=ConnectionMessage::notEqualPasswords;
+  }
+
+  if ( i_user==userInDatabase || i_user.length()==0) {
+    signUpConnectionState= ConnectionMessage::invalidUser;
+  }
+
+  // ToDo: Set user and password in table.
+  return signUpConnectionState;
+}
+
+Login::ConnectionMessage Login::entry ( const string& i_user, const string& i_password ) {
+  ConnectionMessage signUpConnectionState=ConnectionMessage::goodEntry;
+  // ToDo: Search the user and compare password
+  // ToDo: Query asking for password based on user
+  // These strings are just created for debug process
+  string userInDatabase = "Lol2018";
+  string passInDatabase = "Lol2018";
+
+  if ( i_user != userInDatabase ) {
+    signUpConnectionState= ConnectionMessage::doesntExists;
+  }
+  if ( i_password != passInDatabase && (i_user == userInDatabase) ) {
```

Comment:
why are you checking again (i_user == userInDatabase) ?
Use IF, ELSE IF statements

Reply:
I thought that I can't nest if statements.

Reply:
You are right, change applied.